src/main/webapp/src/app/calendar-task/task.component.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnInit} from '@angular/core';
+
+@Component({
+  selector: 'calendar-tasks',
+  templateUrl: './task.component.html',
+  styleUrls: ['./task.component.css', '../common/growpod-page-styles.css'],
+})
+export class TaskComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  /**
+   * Clear tasks before adding any new elements
+   */
+  clearTasks(): void {
+    const tasksContainer = document.getElementById('all-tasks');
+    while (tasksContainer.firstChild) {
+      tasksContainer.removeChild(tasksContainer.lastChild);
+    }
+  }
+
+  /**
```

Comment:
No obligation to change, but: Any reason why you chose not to do this in the Angular template, using `*ngFor` and `*ngIf`?

Reply:
Originally I wanted to use a directive as suggested in the Angular docs, but the way I started implementing the API was by having the schedule component call the gapi session and then from that session calling a directive but this only led to errors about providers and I decided to do the next best thing, which was having a component instead of a directive. I'll definitely keep this comment in mind though; it will be an interesting PR in the future.